### PR TITLE
docs: reorganize README.md

### LINKED ISSUE #1036: README.md is a bit confusing
1. There are two sections in [README.md](https://github.com/deepmodeling/abacus-develop#table-of-contents) named Features, which focus on different content. 
2. There is some confusion about the relationship between [Functionalities](https://github.com/deepmodeling/abacus-develop#functionalities) and [Examples](https://github.com/deepmodeling/abacus-develop#examples). The content links contained in Functionalities point to the contents of Examples.
3. The link contained in [Examples](https://github.com/deepmodeling/abacus-develop#examples) `here` is not working.

### PR DESCRIPTION
**Pull request by ouqi0711**
The documentation has been reorganized and adjusted. The miscellaneous part might need more discussion.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,17 +154,54 @@ ABACUS provides a wide variety of functionalities, with explanation and examples
...
+- Peking University (Dr. Mohan Chen)
+- Institute of Physics, Chinese Academy of Sciences (Dr. Xinguo Ren)
+- AI for Science Institute
 
 [back to top](#readme-top)
 
-# For developers
+# Communicating and making contributions
 
 We also provide some [information](docs/CONTRIBUTING.md) on how to make contributions to ABACUS.
```
Comment (caic99) [inline README.md]:
```suggestion
If you find a bug or have some questions, please refer to our GitHub [issue tracker](https://github.com/deepmodeling/abacus-develop/issues), and our developers are willing to help. We also provide guidelines on how to make contributions to ABACUS.
```

File: README.md [hunk 2]
```diff
@@ -17,36 +17,41 @@ ABACUS is an electronic structure package based on density functional theory(DFT
...
+11. Mulliken charge analysis (only for LCAO).
+12. Projected density of states (PDOS) (only for LCAO).
+13. DFT+U calculation (only for LCAO).
+14. Solvation model method for solvation energy.
+15. Stochastic DFT (only for PW).
+16. DeePKS method (trial version, only for LCAO).
+17. Electric field and dipole correction.
+18. Orbital-free DFT.
+19. (subsidiary tool)Plot_tools for plot PDOS and PBANDS.
+20. (subsidiary tool)Generator for second second generation numerical orbital basis.
```
Comment (hongriTianqi) [inline README.md]: double "second"

Comment (caic99) [reply README.md]:
```suggestion
20. (subsidiary tool)Generator for second generation numerical orbital basis.
```

File: README.md [hunk 3]
```diff
@@ -17,36 +17,41 @@ ABACUS is an electronic structure package based on density functional theory(DFT
...
 
 ABACUS provides the following features and functionalities:
 
-1. Ground-state total energy calculations using Kohn-Sham (KS) density functional theory
-(DFT) with local-density, generalized gradient approximations (LDA/GGAs), and hybrid functionals
-(PBE0 and HSE06, only for LCAO).
-2. Brillouin zone sampling using the Monkhorst-Pack special k-points.
-3. Geometry relaxations with Conjugated Gradient (CG) and BFGS methods.
+1. Three types of supported basis sets: pw, LCAO, and LCAO-in-pw.
+2. Ground-state total energy calculations using Kohn-Sham (KS) density functional theory (DFT) with local-density, generalized gradient approximations (LDA/GGAs), Meta-GGA(depend on LIBXC, only for PW), and hybrid functionals (PBE0 and HSE06, only for LCAO and currently under test).
```
Comment (caic99) [inline README.md]:
```suggestion
2. Ground-state total energy calculations using Kohn-Sham (KS) density functional theory (DFT) with local-density, generalized gradient approximations (LDA/GGAs), Meta-GGA(requires LIBXC, only for PW), and hybrid functionals (PBE0 and HSE06, only for LCAO and currently under test).
```

File: README.md [hunk 4]
```diff
@@ -17,36 +17,41 @@ ABACUS is an electronic structure package based on density functional theory(DFT
...
+3. Geometry relaxations with Conjugated Gradient (CG) and BFGS methods and FIRE methods.
```
Comment (caic99) [inline README.md]:
```suggestion
3. Geometry relaxations with Conjugated Gradient (CG), BFGS, and FIRE methods.
```

File: README.md [hunk 5]
```diff
@@ -17,36 +17,41 @@ ABACUS is an electronic structure package based on density functional theory(DFT
...
-10. Electrostatic potential.
-11. Mulliken charge analysis.
-12. Projected density of states (PDOS).
+10. Print-out of the electrostatic potential.
+11. Mulliken charge analysis (only for LCAO).
+12. Projected density of states (PDOS) (only for LCAO).
+13. DFT+U calculation (only for LCAO).
+14. Solvation model method for solvation energy.
+15. Stochastic DFT (only for PW).
+16. DeePKS method (trial version, only for LCAO).
```
Comment (caic99) [inline README.md]: "trial version" sounds like you should pay for it after 7 days 😂
```suggestion
16. DeePKS method (under development, only for LCAO).
```

File: README.md [hunk 6]
```diff
@@ -64,32 +69,30 @@ Please refer to the [installation guide](docs/install.md) for instruction on the
...
 
-    When performing calculations with numerical atomic orbital basis, it is necessary to prepare a numerical orbital file for each element in the system. Generally, the numerical orbital file should be prepared by the user, which will be described later. The filename for each element’s numerical orbital basis needs to be specified in the `STRU` file. However, in case that the numerical orbital files are stored in a location different from the working directory, then a full path to access the orbital files have to be specified in the `STRU` file.
+    This part is only required in LCAO calculations. 
+    The filename for each element’s numerical orbital basis needs to be specified in the STRU file, together with the directory of the orbital files unless they are already present in the working directory.
     ABACUS provides numerical atomic basis sets of different accuracy levels for most elements commonly used. Users can download these basis sets from the [website](http://abacus.ustc.edu.cn/pseudo.html). Moreover, users can generate numerical atomic orbitals by themselves, and the procedure is provided in this [short introduction](docs/generate-basis.md).
```
Comment (caic99) [inline README.md]: @dyzheng dead link here
http://abacus.ustc.edu.cn/pseudo.html

Comment (ouqi0711) [reply README.md]: https://abacus.ustc.edu.cn/pseudo/list.htm
#should be this link but idk how to make commit on top of this conversation lol

Comment (caic99) [reply README.md]:
```suggestion
    ABACUS provides numerical atomic basis sets of different accuracy levels for most elements commonly used. Users can download these basis sets from the [website](http://abacus.ustc.edu.cn/pseudo/list.htm). Moreover, users can generate numerical atomic orbitals by themselves, and the procedure is provided in this [short introduction](docs/generate-basis.md).
```

File: README.md [hunk 7]
```diff
@@ -164,19 +156,56 @@ ABACUS provides a wide variety of functionalities, with explanation and examples
...
+    [1]. Qianrui Liu, and Mohan Chen. "Plane-Wave-Based Stochastic-Deterministic Density Functional Theory for Extended Systems." https://arxiv.org/abs/2204.05662
+    
+- **If DFT+U is used:**
+
+    [1]. Xin Qu, et al. "DFT+ U within the framework of linear combination of numerical atomic orbitals." The Journal of Chemical Physics (2022).
 
-In the directory tests/, each sub-directory contains a separate test example. An introduction of the examples in tests/ directory can be found [here](tests/README.md). In each subdirectory, you may also find a file named jd which contains a short job description, and for some cases you may also find a README file containing more details about the run. Also, reference output is provided in the file `result.ref`.
+- **If second generation numerical orbital basis is used:**
+
+    [1]. Peize Li, Xinguo Ren, and Lixin He. "Strategy for constructing compact numerical atomic orbital basis sets by incorporating the gradients of reference wavefunctions." Physical Review B 103.23 (2021): 235131.
```
Comment (pxlxingliang) [inline README.md]: Should be "Peize Lin"

Comment (ouqi0711) [reply README.md]:
```suggestion
    [1]. Peize Lin, Xinguo Ren, and Lixin He. "Strategy for constructing compact numerical atomic orbital basis sets by incorporating the gradients of reference wavefunctions." Physical Review B 103.23 (2021): 235131.
```

**Review by pxlxingliang — APPROVED** [review]

**Pull request merged**